python/ray/serve/pipeline/generate.py

```diff
@@ -51,41 +51,19 @@ def reset(cls):
             cls.__shared_state = dict()
 
 
-def _remove_non_default_ray_actor_options(ray_actor_options: Dict[str, Any]):
-    """
-    In Ray DAG building we pass full ray_actor_options regardless if a field
-    was explicitly set. Since some values are invalid, we need to remove them
-    from ray_actor_options.
-    """
-    # TODO: (jiaodong) Revisit when we implement build() when user explicitly
-    # pass default value
-    ray_actor_options = {k: v for k, v in ray_actor_options.items() if v}
-    if ray_actor_options.get("placement_group") == "default":
-        del ray_actor_options["placement_group"]
-    if ray_actor_options.get("placement_group_bundle_index") == -1:
-        del ray_actor_options["placement_group_bundle_index"]
-    if ray_actor_options.get("max_pending_calls") == -1:
-        del ray_actor_options["max_pending_calls"]
-
-    return ray_actor_options
-
-
 def transform_ray_dag_to_serve_dag(dag_node):
     """
     Transform a Ray DAG to a Serve DAG. Map ClassNode to DeploymentNode with
     ray decorated body passed in, ans ClassMethodNode to DeploymentMethodNode.
     """
     if isinstance(dag_node, ClassNode):
         deployment_name = DeploymentNameGenerator.get_deployment_name(dag_node)
-        ray_actor_options = _remove_non_default_ray_actor_options(
```

Comment:
I'm not sure why we need this in the first place, but we should check with the author to make sure removing this is fine.

Reply:
This is because the design flaw in the original `.options` sends garbage placeholder values here. Now this code does not take effects any more after #23127.. @jiaodong for further confirmation.